### PR TITLE
added SUSE and openSUSE

### PR DESCRIPTION
**Pull request by msmeissn**
Added the SUSE and openSUSE ecosystem.

https://github.com/ossf/osv-schema/issues/259

### CODE REVIEW

File: docs/schema.md [hunk 1]
```diff
@@ -387,6 +387,17 @@ The defined database prefixes and their "home" databases are:
...
+      <td><code>SUSE and openSUSE</code></td>
+      <td><a href="https://www.suse.com/support/security/">SUSE Security Landing page</a></td>
+      <td>
+        <ul>
+          <li>How to contribute: <a href="https://www.suse.com/support/security/contact/">https://www.suse.com/support/security/contact/</a></li>
+          <li>Source URL: <code>https://www.suse.com/support/update/</code></li>
+          <li>OSV Formatted URL: <code>https://ftp.suse.com/pub/projects/security/osv/</code></li>
+        </ul>
+      </td>
+    </tr>
```
Comment (oliverchang) [inline docs/schema.md:400]: Would you be able to add some ecosystem definitions here: https://github.com/ossf/osv-schema/blob/516347895e3541e04db8166f9c83a22250f5b44d/docs/schema.md#affectedpackage-field also ?

There's a few other Linux distros in there that could potentially be useful to base off.

**Review by oliverchang — COMMENTED** [review]
awesome, thank you!

File: validation/schema.json [hunk 2]
```diff
@@ -300,7 +300,7 @@
       "type": "string",
       "title": "Currently supported ecosystems",
       "description": "These ecosystems are also documented at https://ossf.github.io/osv-schema/#affectedpackage-field",
-      "pattern": "^(AlmaLinux|Alpine|Android|Bioconductor|Bitnami|ConanCenter|CRAN|crates.io|Debian|GHC|GitHub Actions|GIT|Go|Hackage|Hex|Linux|Maven|npm|NuGet|OSS-Fuzz|Packagist|Photon OS|Pub|PyPI|Red Hat|Rocky Linux|RubyGems|SwiftURL|Ubuntu)(:[[:digit:]]+)?"
+      "pattern": "^(AlmaLinux|Alpine|Android|Bioconductor|Bitnami|ConanCenter|CRAN|crates.io|Debian|GHC|GitHub Actions|GIT|Go|Hackage|Hex|Linux|Maven|npm|NuGet|openSUSE|OSS-Fuzz|Packagist|Photon OS|Pub|PyPI|Red Hat|Rocky Linux|RubyGems|SUSE|SwiftURL|Ubuntu)(:[[:digit:]]+)?"
     },
     "prefix": {
```
Comment (andrewpollock) [inline validation/schema.json:305]: The prefix(es) (`SUSE-SU`?) need(s) to be added to the validation here.

File: docs/schema.md [hunk 3]
```diff
@@ -387,6 +387,17 @@ The defined database prefixes and their "home" databases are:
         </ul>
       </td>
     </tr>
+    <tr>
+      <td><code>SUSE and openSUSE</code></td>
```
Comment (andrewpollock) [inline docs/schema.md:391]: This needs to be the record prefix, which from a brief eyeball of https://lists.suse.com/pipermail/sle-security-updates/ should be `SUSE-SU`?

```suggestion
      <td><code>SUSE-SU</code></td>
```

**Review by andrewpollock — CHANGES_REQUESTED** [review]

File: validation/schema.json (hunk 2 above)
Comment (msmeissn) [reply validation/schema.json:305]: not in the expression below?

File: docs/schema.md (hunk 1 above)
Comment (msmeissn) [reply docs/schema.md:411]: I added stuff there

File: validation/schema.json (hunk 2 above)
Comment (msmeissn) [reply validation/schema.json:305]: i added in the expression below

File: docs/schema.md (hunk 3 above)
Comment (msmeissn) [reply docs/schema.md]: changed, with and not in code

File: docs/schema.md [hunk 4]
```diff
@@ -685,13 +696,15 @@ The defined ecosystems are:
 | `npm` | The NPM ecosystem; the `name` field is an NPM package name.  |
 | `NuGet` | The NuGet package ecosystem. The `name` field is a NuGet package name.  |
 | `OSS-Fuzz` | For reports from the OSS-Fuzz project that have no more appropriate ecosystem; the `name` field is the name assigned by the OSS-Fuzz project, as recorded in the submitted fuzzing configuration.  |
+| `openSUSE` | The openSUSE ecosystem; The ecosystem specifier is the marketing name of the openSUSE distribution. With the `:` removed, it matches the value in the /etc/os-release `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is a `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific openSUSE distribution. The ECOSYSTEM version ordering is the RPM versioncompare ordering, and the database uses the `introduced` and `fixed` boundaries.|
```
Comment (andrewpollock) [inline docs/schema.md:699]: Minor grammatical and styling:

```suggestion
| `openSUSE` | The openSUSE ecosystem; The ecosystem specifier is the marketing name of the openSUSE distribution. With the `:` removed, it matches the value in the `/etc/os-release` `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is an `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific openSUSE distribution. The ECOSYSTEM version ordering is the RPM versioncompare ordering, and the database uses the `introduced` and `fixed` boundaries.|
```

**Review by andrewpollock — APPROVED** [review]
LGTM with some minor style and grammar

File: docs/schema.md (hunk 4 above)
Comment (oliverchang) [inline docs/schema.md:699]:
```suggestion
| `openSUSE` | The openSUSE ecosystem; The ecosystem string has a `:<RELEASE>` suffix representing the marketing name of the openSUSE distribution. `<RELEASE>` matches the value in the /etc/os-release `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is a `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific openSUSE distribution. The ECOSYSTEM version ordering is the RPM versioncompare ordering, and the database uses the `introduced` and `fixed` boundaries.|
```

Comment (oliverchang) [reply docs/schema.md:699]: (One more suggestion for consistency with other ecosystem definitions)

File: docs/schema.md [hunk 5]
```diff
@@ -685,13 +696,15 @@ The defined ecosystems are:
...
 | `Rocky Linux` | The Rocky Linux package ecosystem; the `name` is the name of the source package. The ecosystem string might optionally have a `:<RELEASE>` suffix to scope the package to a particular Rocky Linux release. `<RELEASE>` is a numeric version.
 | `RubyGems` | The RubyGems ecosystem; the `name` field is a gem name.  |
+| `SUSE` | The SUSE ecosystem; The ecosystem specifier is the marketing name of the SUSE product. With the `:` removed, it matches the value in the /etc/os-release `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is a `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific SUSE product. The ECOSYSTEM version ordering is the RPM versioncompare ordering, and the database uses the `introduced` and `fixed` boundaries.|
```
Comment (oliverchang) [inline docs/schema.md:707]:
```suggestion
| `SUSE` | The SUSE ecosystem; The ecosystem string has a `:<RELEASE>` suffix representing the marketing name of the SUSE product. `<RELEASE>` matches the value in the /etc/os-release `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is a `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific SUSE product. The ECOSYSTEM version ordering is the RPM versioncompare ordering, and the database uses the `introduced` and `fixed` boundaries.|
```

**Review by oliverchang — COMMENTED** [review]
thanks!

**Comment by msmeissn** [comment]
Thanks. I had to resolve conflicts with the last ubuntu change, and squashed.

I also changed the wording again a bit, as we have "SUSE:Linux Enterprise Server" ... so the prefix includes part of the marketing name.

File: docs/schema.md [hunk 6]
```diff
@@ -696,13 +707,15 @@ The defined ecosystems are:
 | `npm` | The NPM ecosystem; the `name` field is an NPM package name.  |
 | `NuGet` | The NuGet package ecosystem. The `name` field is a NuGet package name.  |
 | `OSS-Fuzz` | For reports from the OSS-Fuzz project that have no more appropriate ecosystem; the `name` field is the name assigned by the OSS-Fuzz project, as recorded in the submitted fuzzing configuration.  |
+| `openSUSE` | The openSUSE ecosystem; The ecosystem specifier is the marketing name of the openSUSE distribution. With the `:` removed, it matches the value in the `/etc/os-release` `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is an `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific openSUSE distribution. The ECOSYSTEM version ordering is the RPM versioncompare ordering, and the database uses the `introduced` and `fixed` boundaries.|
```
Comment (oliverchang) [inline docs/schema.md:710]: Would you be able to make this consistent with the text for `SUSE` ? i.e. "The ecosystem string has a `:<RELEASE>` suffix representing the marketing name of the openSUSE distribution..."

**Review by oliverchang — COMMENTED** [review]
thanks! Just one last comment.

File: docs/schema.md (hunk 6 above)
Comment (msmeissn) [reply docs/schema.md]: done so

**Review by oliverchang — APPROVED** [review]

**Pull request merged**